### PR TITLE
Harvest: deposit multiple assets supplying multiple assets

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR implements functionality required to meet the acceptance criteria of https://app.asana.com/0/1156716699555540/1198904860690582/f.
- Support borrowing multiple assets in a single `Borrow` transaction
- Subtract value previously borrowed from user's current borrowable amount

Additional changes
- Introduce local cache for MoneyMarket params to reduce param reads

### CODE REVIEW

File: x/harvest/keeper/borrow.go [hunk 1]
```diff
@@ -40,66 +39,85 @@ func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, coins sdk.Coins
...
+	// Get the total borrowable USD amount at user's existing deposits
 	deposits := k.GetDepositsByUser(ctx, borrower)
 	if len(deposits) == 0 {
 		return sdkerrors.Wrapf(types.ErrDepositsNotFound, "no deposits found for %s", borrower)
 	}
 	totalBorrowableAmount := sdk.ZeroDec()
 	for _, deposit := range deposits {
-		borrowableAmountForDeposit, err := k.getBorrowableAmountForDeposit(ctx, deposit)
+		// Fetch money market and store in local cache
+		if moneyMarketCache[deposit.Amount.Denom] == (types.MoneyMarket{}) {
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:74]: ditto

File: x/harvest/keeper/borrow.go [hunk 2]
```diff
@@ -40,66 +39,85 @@ func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, coins sdk.Coins
...
+	existingBorrow, found := k.GetBorrow(ctx, borrower)
 	if found {
-		// TODO: here we're assuming that the user only has 1 previous borrow. To be addressed in future cards.
-		previousBorrow := previousBorrows.Amount[0]
-		previousBorrowUSDValue, err := k.calculateUSDValue(ctx, previousBorrow.Amount, previousBorrow.Denom)
-		if err != nil {
-			return err
+		for _, borrowedCoin := range existingBorrow.Amount {
+			// Fetch money market and store in local cache
+			if moneyMarketCache[borrowedCoin.Denom] == (types.MoneyMarket{}) {
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:99]: ditto

File: x/harvest/keeper/borrow_test.go [hunk 3]
```diff
@@ -27,8 +29,11 @@ func (suite *KeeperTestSuite) TestBorrow() {
 		loanToValueKAVA           sdk.Dec
 		priceBTCB                 sdk.Dec
 		loanToValueBTCB           sdk.Dec
+		priceBNB                  sdk.Dec
```
Comment (karzak) [inline x/harvest/keeper/borrow_test.go:32]: These args are pretty verbose, could perhaps be simplified with a struct. Not a blocker though imo.

**Review by karzak — CHANGES_REQUESTED** [review]
Looking good

File: x/harvest/keeper/borrow_test.go (hunk 3 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow_test.go:32]: Yeah, I'm planning on touching up the test config once we've got these initial tasks through to acceptance.

File: x/harvest/keeper/borrow.go (hunk 2 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/712/commits/0f73cb5c167ae2281214afcde17a065e779082f4

File: x/harvest/keeper/borrow.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/712/commits/0f73cb5c167ae2281214afcde17a065e779082f4

File: x/harvest/keeper/borrow.go [hunk 4]
```diff
@@ -40,66 +39,85 @@ func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, coins sdk.Coins
...
-	proprosedBorrowUSDValue, err := k.calculateUSDValue(ctx, amount.Amount, amount.Denom)
-	if err != nil {
-		return err
+func (k Keeper) ValidateBorrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coins) error {
+	// Get the proposed borrow USD value
+	moneyMarketCache := map[string]types.MoneyMarket{}
+	proprosedBorrowUSDValue := sdk.ZeroDec()
+	for _, coin := range amount {
+		// Fetch money market and store in local cache
+		if moneyMarketCache[coin.Denom] == (types.MoneyMarket{}) {
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/712/commits/0f73cb5c167ae2281214afcde17a065e779082f4

**Review by karzak — APPROVED** [review]

**Pull request merged**